packages/frontend/src/features/sqlRunner/components/SaveCustomViewModal.tsx

```diff
@@ -0,0 +1,245 @@
+                        multiline
+                        maw={300}
+                        label="Custom explores are a way to save a query that
+                                you can reuse later when 'Querying from tables'.
+                                You can use them to save time when you and your
+                                organization want to run the same query again."
+                    >
+                        <MantineIcon
+                            color="gray.7"
+                            icon={IconInfoCircle}
+                            size={16}
+                        />
+                    </Tooltip>
+                </Group>
+            }
+            styles={(theme) => ({
+                header: { borderBottom: `1px solid ${theme.colors.gray[4]}` },
+                body: { padding: 0 },
+            })}
+        >
+            <form onSubmit={form.onSubmit(handleSubmit)}>
+                <Stack p="md">
+                    <Stack spacing="xs">
+                        <TextInput
+                            radius="md"
+                            label="Name"
+                            required
+                            {...form.getInputProps('name')}
+                        />
+                    </Stack>
+
+                    <Radio.Group
+                        label="Custom view type"
+                        {...form.getInputProps('customViewType')}
+                    >
+                        <Group mt="md">
+                            <Radio
+                                size="xs"
+                                value={CustomViewType.VIRTUAL}
+                                label="Create virtual custom view"
+                            />
+                            <Radio
+                                size="xs"
+                                disabled={!canWriteToDbtProject}
+                                value={CustomViewType.PERSISTENT}
+                                label="Write back to dbt project"
+                            />
+                        </Group>
+                    </Radio.Group>
+
+                    <Alert>
+                        <Text fz="xs">
+                            {getCustomViewTypeAlertText(
+                                form.values.customViewType,
+                            )}
+                        </Text>
+                    </Alert>
+
+                    {form.values.customViewType ===
+                        CustomViewType.PERSISTENT && (
+                        <Stack spacing="xs" pl="xs">
+                            <Text fw={500}>
+                                Files to be created in{' '}
+                                <Badge
+                                    radius="md"
+                                    variant="light"
+                                    color="gray.9"
+                                    fz="xs"
+                                >
+                                    REPO NAME
```

Comment:
We can build a new endpoint to get this and the file path , or you can take this from `api/v1/github/repos/list`

Reply:
We could make this a Link that opens the Repo in a separate tab

Reply:
Added a TODO